src/Casdoor.Client/CasdoorClient.Internal.cs

```diff
@@ -13,13 +13,33 @@
 // limitations under the License.
 
 using System.Net.Http.Json;
+using System.Text.Json;
 
 namespace Casdoor.Client;
 
 public partial class CasdoorClient
 {
-    internal Task<TValue?> GetFromJsonAsync<TValue>(string? requestUri, CancellationToken cancellationToken = default)
-        => _httpClient.GetFromJsonAsync<TValue>(requestUri, cancellationToken);
+    internal async Task<TValue?> GetFromJsonAsync<TValue>(string? requestUri, CancellationToken cancellationToken = default)
+    {
+        var response = await _httpClient.GetAsync(requestUri, cancellationToken);
+        using var stream = await response.Content.ReadAsStreamAsync();
+        TValue? successResult;
+        try
+        {
+            successResult = await JsonSerializer.DeserializeAsync<TValue>(stream, cancellationToken: cancellationToken);
+        }
+        catch (JsonException e)
+        {
+            stream.Seek(0, SeekOrigin.Begin);
```

Comment:
It looks like a crutch - at first we try to read some type from the stream, then if we caught Json exception we try to rewind stream and deserialize it as Casdoor response. I guess it's a bit awkward - why don't we throw immediately with serialization error showing SDK user about API and SDK mismatches instead of vague errors that casdoor returned?

Reply:
As mentioned in the [issue#38](https://github.com/casdoor/casdoor-dotnet-sdk/issues/38), serialization exceptions in json are exceptions that occur in the program itself, and incorrect tokens, etc. are what cause them to occur. The original exception message does not help to locate the root cause. But as you said, the information in the msg field is still not precise enough, I guess we can throw the json exception and the msg content at the same time?

Reply:
> As mentioned in the https://github.com/casdoor/casdoor-dotnet-sdk/issues/38, serialization exceptions in json are exceptions that occur in the program itself, and incorrect tokens, etc. are what cause them to occur.

Correct and that is the problem.

> The original exception message does not help to locate the root cause.

You are correct again and the problem in the case i suppose our server response expectations and real answer are different. For example - we expect to get CasdoorOrganization but server respond by CasdoorResponse - this case response can be ok and no error occured during request processing so deserialize response as CasdoorResponse in case of our expecting CasdoorOrganization is useless and will vague us more because root cause is SDK and server response mismatch.

So in this case i would throw CasdoorApiException as you did without trying to deserialize response once again as CasdoorResponse because it maybe correct one where Msg property will be empty and this will vague us more.

Example
```csharp
try
{
    successResult = await JsonSerializer.DeserializeAsync<TValue>(stream, cancellationToken: cancellationToken);
}
catch (JsonException e)
{
	throw new CasdoorApiException("Server response cannot be deserialized as TValue. Server API and SDK implementation are inconsistent", e);
}
return successResult;
```

Reply:
Thanks, this is much clearer! I will modify the code.

Reply:
The error msg is "Unauthorized operation" at the #38 sample, I think this msg needs to tell to the user. It is a normal and not a code implementation error, it will also throw when your token is invalid.
We can check the code field to determine how to throw the exception. I think it is better to return `JsonException` with the msg when the response code is not an error because I think it is more helpful to clearly distinguish `CasdoorApiException` and other exceptions,  `CasdoorApiException` is only thrown when casdoor returns the error code.

Reply:
I think if an operation is Unauthorized operation API should response using 403 http code instead of doing it using normal flow but with different response structrure or always return same (universal) JSON that contans useful data inside one of the properties Data, Data2 etc. i.e. CasdoorResponse

Reply:
@aegoroff Yes, you are right, returned 401/403 HTTP code or all APIs using the base response is better. Maybe we can open an issue with Casdoor.

Reply:
I found an issue like this but a bit more wider (https://github.com/casdoor/casdoor/issues/2065) but it had been closed

Reply:
@aegoroff we use Response.status instead of HTTP status because we won't want to mess with API gateway. Casdoor is business logic. Casdoor should not mess with HTTP status. If there's a firewall or API gatway before Casdoor, it can mess with HTTP status, which has nothing to do with Casdoor.

Reply:
@hsluoyz interesting point of view - never though about it. Are there many cases where API gateway or firewall mess http statuses of underlying services?